src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationContext.java

```diff
@@ -369,6 +386,7 @@ public CppModuleMap getVerificationModuleMap() {
     private boolean propagateModuleMapAsActionInput = true;
     private CppConfiguration.HeadersCheckingMode headersCheckingMode =
         CppConfiguration.HeadersCheckingMode.STRICT;
+    private Map<String, String> virtualToOriginalHeaders = new HashMap<>();
```

Comment:
Let's make this at least LinkedHashMap, or maybe even better ImmutableMap.Builder? To keep the order of files deterministic?

Reply:
Hmm now I'm thinking, if we should make it a NestedSet of Pairs instead, to not be quadratic in used memory, and then create the Map only when actually iterating the map. Wdyt?